### ISSUE TITLE
Get rid of the `random_state` argument in `get_pipeline`

### ISSUE THREAD

**Issue by freddyaboulton** [alteryx/evalml#1638]
Currently, the `best_pipeline` property in `AutoMLSearch` will [always return](https://github.com/alteryx/evalml/blob/main/evalml/automl/automl_search.py#L530) a pipeline with `random_state=0` because that's the default value for `random_state` in the `get_pipeline` method.

I think this is confusing because if I create an `AutoMLSearch` object with `random_state=5`, for example, I expect all of the pipelines created and returned by `AutoMLSearch` to have `random_state=5`.

I think we should remove the `random_state` argument from `get_pipeline` and instead make sure that the random state is set to whatever it was when the pipeline was fit.

**Comment by chukarsten** [comment]
Agreed.  I think that's the right move @freddyaboulton.  This one doesn't seem like a huge lift.  I'll dev backlog it.  @dsherry @tyler3991

**Comment by bchen1116** [comment]
@freddyaboulton @chukarsten @dsherry after looking at this one for a little, I think this issue might be a not be as straightforward as expected. In our `pipeline_base.py` [class](https://github.com/alteryx/evalml/blob/4ff4aae32f6d084949eea3c3c15a7f2beaea2b59/evalml/pipelines/pipeline_base.py#L63), we set the `random_state` to be an `np.RandomState` object, which we generate through our utils, `get_random_state`, which uses [sklearn.utils.check_random_state](https://scikit-learn.org/stable/modules/generated/sklearn.utils.check_random_state.html). While this state is the same for the same seed, we can't get the original seed we passed in. This means that even if we passed in `random_state=4` for AutoMLSearch, the pipelines and internal `random_state` values are np.RandomState objects seeded with '4'. 

I can think of three possible solutions for this issue:
- We can continue to remove `random_state` argument, but the `random_state` associated with the pipeline continues to be a np.RandomState object. This might be confusing for the user, but currently, the `random_state` value for `AutoMLSearch` is also an np.RandomState object
![image](https://user-images.githubusercontent.com/22552445/105252291-25d01200-5b4b-11eb-97d9-bc1b7e854c5c.png)
- We can introduce a `random_seed` value like we did with AutoMLSearch, but I'm not sure this would help much
- We can try to override np.RandomState with the int provided instead, and keep the `random_state` value as an int instead of an `np.RandomState` object. This makes the values consistent with what the user expects, but might be a lot of work, since we'll likely have to refactor both pipelines and components.

I think the big issue is determining why we decided to use `np.RandomState` objects to hold the random state of our objects. Was there a reason we chose to pass this around versus an int?

What do you all think? I think option 1 would be the best, even if it introduces some potential inconsistency with what the user expects.

**Comment by bchen1116** [comment]
@freddyaboulton I think the largest issue is that in our [pipeline_base](https://github.com/alteryx/evalml/blob/4ff4aae32f6d084949eea3c3c15a7f2beaea2b59/evalml/pipelines/pipeline_base.py#L63), we set the `random_state` to be an np.RandomState object, as seen here
![image](https://user-images.githubusercontent.com/22552445/105392860-af362180-5be9-11eb-8b23-52e80c359265.png)


If we remove the `np.RandomState` object, then we can preserve the random state that the user could pass in, but we would also need to change the default `random_state` behavior for [component_base](https://github.com/alteryx/evalml/blob/4ff4aae32f6d084949eea3c3c15a7f2beaea2b59/evalml/pipelines/components/component_base.py#L29) as well. This issue does seem to be tied with a discussion about support for `np.RandomState`. What do you think?

**Comment by bchen1116** [comment]
Through discussion with @freddyaboulton, we decided it's fine to have the pipeline `random_state` be a `np.RandomState` object, but we want to make sure it's seeded with the `random_state` int that a user passes in, OR with the `np.RandomState` object that a user passes in, depending on the input. 

This means that if a user passes in `random_state=4` to `AutoMLSearch`, we create the pipeline `random_state` with '4' as the seed, and if a user passes in `random_state=np.RandomState()`, we pass that to create the pipeline instead.
